src/widgets/numeric-menu/numeric-menu.tsx

```diff
@@ -47,77 +50,92 @@ const renderer = ({
- * @property {number} [end] High bound of the option (<=).
- */
-
-/**
- * @typedef {Object} NumericMenuWidgetOptions
- * @property {string|HTMLElement} container CSS Selector or HTMLElement to insert the widget.
- * @property {string} attribute Name of the attribute for filtering.
- * @property {NumericMenuOption[]} items List of all the items.
- * @property {NumericMenuTemplates} [templates] Templates to use for the widget.
- * @property {NumericMenuCSSClasses} [cssClasses] CSS classes to add to the wrapping elements.
- * @property {function(object[]):object[]} [transformItems] Function to transform the items passed to the templates.
- */
-
-/**
- * The numeric menu is a widget that displays a list of numeric filters in a list. Those numeric filters
- * are pre-configured with creating the widget.
- *
- * @requirements
- * The attribute passed to `attribute` must be declared as an [attribute for faceting](https://www.algolia.com/doc/guides/searching/faceting/#declaring-attributes-for-faceting) in your
- * Algolia settings.
- *
- * The values inside this attribute must be JavaScript numbers and not strings.
- *
- * @type {WidgetFactory}
- * @devNovel NumericMenu
- * @category filter
- * @param {NumericMenuWidgetOptions} $0 The NumericMenu widget items
- * @return {Widget} Creates a new instance of the NumericMenu widget.
- * @example
- * search.addWidgets([
- *   instantsearch.widgets.numericMenu({
- *     container: '#popularity',
- *     attribute: 'popularity',
- *     items: [
- *       { label: 'All' },
- *       { end: 500, label: 'less than 500' },
- *       { start: 500, end: 2000, label: 'between 500 and 2000' },
- *       { start: 2000, label: 'more than 2000' }
- *     ]
- *   })
- * ]);
- */
-export default function numericMenu({
-  container,
-  attribute,
-  items,
-  cssClasses: userCssClasses = {},
-  templates = defaultTemplates,
-  transformItems,
-} = {}) {
+export type NumericMenuCSSClasses = {
+  /**
+   * CSS class to add to the root element.
+   */
+  root?: string | string[];
+
+  /**
+   * CSS class to add to the root element when no refinements.
+   */
+  noRefinementRoot?: string | string[];
+
+  /**
+   * CSS class to add to the list element.
+   */
+  list?: string | string[];
+
+  /**
+   * CSS class to add to each item element.
+   */
+  item?: string | string[];
+
+  /**
+   * CSS class to add to each selected item element.
+   */
+  selectedItem?: string | string[];
+
+  /**
+   * CSS class to add to each label element.
+   */
+  label?: string | string[];
+
+  /**
+   * CSS class to add to each label text element.
+   */
+  labelText?: string | string[];
+
+  /**
+   * CSS class to add to each radio element (when using the default template).
+   */
+  radio?: string | string[];
+};
+
+export type NumericMenuTemplates = {
+  /**
+   * Item template, provided with `label` (the name in the configuration), `isRefined`, `url`, `value` (the setting for the filter) data properties.
+   */
+  item?: Template;
```

Comment:
We can type the template [as described in the documentation](https://www.algolia.com/doc/api-reference/widgets/numeric-menu/js/#widget-param-item) since it's generic.

Reply:
@francoischalifour Can you rephrase what are you asking for? Sorry, didn't got it.

Reply:
something roughly like this?

```
Template<{
  /**
   * the name of the attribute.
   */
  attribute: string;
  /**
   *  the label for the option.
   */
  label: string;
  /**
   * the encoded URL of the bounds object with a {start, end} form. This value can be used verbatim in the webpage and can be read by refine directly. If you want to inspect the value, you can do JSON.parse(window.decodeURI(value)) to get the object.
   */
  value: string;
  /**
   *  whether or not the refinement is selected.
   */
  isRefined: boolean;
  /**
   * the URL with the applied refinement.
   */
  url: string;
  /**
   * the CSS classes provided to the widget.
   */
  cssClasses: object;
}>
```

Reply:
@Haroenv Can you confirm that this type used in this generic, is no used elsewhere?

Reply:
@Haroenv Can you check the contents now?